core/src/test/java/org/apache/calcite/test/SqlToRelConverterTest.java

```diff
@@ -2559,6 +2559,18 @@ public void testDynamicStarInTableJoin() throws Exception {
     sql(sql).with(getTesterWithDynamicTable()).ok();
   }
 
+  /**
+   * Test case for <a href="https://issues.apache.org/jira/browse/CALCITE-2900">[CALCITE-2900]
+   * RewriteRexShuttle generates wrong type during flatten for structured type when
+   * visitInputRef</a>.
+   */
+  @Test
+  public void testNestedColumn() {
```

Comment:
@zabetak This is the new a little "low-level" test I added. SqlToRelConverter.flattenTypes is almost the lowest level entry point for this test. The other option is RelStructuredTypeFlattener.rewrite. But for this test, I have to write a lot of boilerplate code for this single test. Since there is almost no extra logic applied in SqlToRelConverter.flattenTypes, I think it should be OK.

Let me know. Thanks!

Reply:
SqlToRelConverterTest is a good place. Actually it seems that you can move all the tests here. Moreover this new test does not bring something significant in comparison with the previous ones so I would suggest to remove it.